src/Parsing/ParserState.php

```diff
@@ -69,15 +64,21 @@ public function __construct($text, Settings $parserSettings, $lineNumber = 1)
     /**
      * Sets the charset to be used if the CSS does not contain an `@charset` declaration.
      *
-     * @param string $charset
+     * @throws SourceException
      */
-    public function setCharset($charset): void
+    public function setCharset(string $charset): void
```

Comment:
I'd prefer to add the native type declaration in a separate PR (so this PR here is more focused).

Reply:
#958